File: app/src/main/java/com/example/services/DownloadThread.java
package com.example.services;

import android.os.Looper;
import android.util.Log;

public class DownloadThread extends Thread {
    private static final String TAG = "MyTag";

    public DownloadHandler mHandler;

    @Override
    public void run() {
        Log.e(TAG, "run: Download Thread : " + Thread.currentThread().getName());

        Looper.prepare();
        mHandler = new DownloadHandler();
        Looper.loop();
    }
}

File: app/src/main/java/com/example/services/DownloadThread.kt
package com.example.services

import java.lang.Thread
import com.example.services.DownloadHandler
import android.util.Log
import com.example.services.DownloadThread
import android.os.Looper

class DownloadThread : Thread() {
    var mHandler: DownloadHandler? = null

    override fun run() {
        Log.e(TAG, "run: Download Thread : " + currentThread().name)
        Looper.prepare()
        mHandler = DownloadHandler()
        Looper.loop()
    }

    companion object {
        private const val TAG = "MyTag"
    }
}